textacy/tm/topic_model.py

```diff
@@ -0,0 +1,294 @@
+                    [(0, ((4, 0.3217), (2, 0.2154)))]
+        """
+        if topics == -1:
+            topics = range(self.n_topics)
+        elif isinstance(topics, int):
+            topics = (topics,)
+
+        for topic_idx in topics:
+            top_doc_idxs = np.argsort(doc_topic_matrix[:, topic_idx])[:-top_n - 1:-1]
+            if weights is False:
+                yield (topic_idx,
+                       tuple(doc_idx for doc_idx in top_doc_idxs))
+            else:
+                yield (topic_idx,
+                       tuple((doc_idx, doc_topic_matrix[doc_idx, topic_idx]) for doc_idx in top_doc_idxs))
+
+    def top_doc_topics(self, doc_topic_matrix, docs=-1, top_n=3, weights=False):
+        """
+        Get the top ``top_n`` topics by weight per doc for ``docs`` in ``doc_topic_matrix``.
+
+        Args:
+            doc_topic_matrix (numpy.ndarray): document-topic matrix with shape
+                (n_docs, n_topics), the result of calling
+                :func:`get_doc_topic_matrix() <textacy.topic_modeling.get_doc_topic_matrix>`
+            docs (seq(int) or int, optional): docs for which to return top topics;
+                if -1, all docs' top topics are returned
+            top_n (int, optional): number of top topics to return per doc
+            weights (bool, optional): if True, docs are returned with their corresponding
+                (normalized) topic weights; otherwise, docs are returned without weights
+
+        Yields:
+            tuple(int, tuple(int)) or tuple(int, tuple(int, float)):
+                next tuple corresponding to a doc; the first element is the doc's
+                index; if ``weights`` is False, the second element is a tuple of ints
+                representing the top ``top_n`` related topics; otherwise, the second
+                is a tuple of (int, float) pairs representing the top ``top_n``
+                related topics and their associated weights wrt the doc; for example::
+
+                    >>> list(TopicModel.top_doc_topics(dtm, docs=(0, 1), top_n=2, weights=False))
+                    [(0, (1, 4)), (1, (3, 2))]
+                    >>> list(TopicModel.top_doc_topics(dtm, docs=0, top_n=2, weights=True))
+                    [(0, ((1, 0.2855), (4, 0.2412)))]
+        """
+        if docs == -1:
+            docs = range(doc_topic_matrix.shape[0])
+        elif isinstance(docs, int):
+            docs = (docs,)
+
+        for doc_idx in docs:
+            top_topic_idxs = np.argsort(doc_topic_matrix[doc_idx, :])[:-top_n - 1:-1]
+            if weights is False:
+                yield (doc_idx,
+                       tuple(topic_idx for topic_idx in top_topic_idxs))
+            else:
+                yield (doc_idx,
+                       tuple((topic_idx, doc_topic_matrix[doc_idx, topic_idx]) for topic_idx in top_topic_idxs))
+
+    def topic_weights(self, doc_topic_matrix):
```

Comment:
Doesn't LDA produce a weighting of topics in the corpus?  Does that give the same results as this?